### PR TITLE
Test Github.Sync.installation/1 event 

### LINKED ISSUE #1373: Write tests for GitHub.Sync.installation_event/1
# Problem

`GitHub.Sync.installation_event/1` is currently completely untested. Since we want to unit test its individual elements to, it should suffice to test each possible outcome of the transaction performed by the tests, as defined by `GitHub.Sync.installation_event_outcome()`

## Subtasks
- [x] write tests for `{:ok, GithubAppInstallation.t()}` outcome
- [ ] write tests for `{:error, :validation_error_on_syncing_installation, Changeset.t()}` outcome
- [x] write tests for `{:error, :multiple_unprocessed_installations_found}` outcome
- [x] write tests for `{:error, :github_api_error_on_syncing_repos, struct}` outcome
- [ ] write tests for `{:error, :validation_error_on_deleting_removed_repos, {list, list}}` outcome
- [ ] write tests for `{:error, :validation_error_on_syncing_existing_repos, {list, list}}` outcome
- [ ] write tests for `{:error, :validation_error_on_marking_installation_processed, Changeset.t()}` outcome
- [ ] write tests for `{:error, :unexpected_transaction_outcome, any}` outcome

### PR DESCRIPTION
**Pull request by zacck-zz**
# What's in this PR?

Added a test block for Github.Sync.installation event 

Make sure any changes to code include changes to documentation.

## References
Fixes #1373

Progress on: #

### CODE REVIEW

**Comment by zacck-zz** [comment]
Some clauses defined seem difficult or impossible to reach so their tests were omitted
    - {:error, :validation_error_on_syncing_installation, Changeset.t()}
    - {:error, :validation_error_on_marking_installation_processed, Changeset.t()}
    - {:error, :unexpected_transaction_outcome, any}
    - {:error, :validation_error_on_deleting_removed_repos, {list, list}}
   - {:error, :validation_error_on_syncing_existing_repos, {list, list}}
  However, if these clauses can be caused by some updates upstream we should cover them with tests

File: test/lib/code_corps/github/sync/sync_test.exs [hunk 1]
```diff
@@ -24,8 +24,49 @@ defmodule CodeCorps.GitHub.SyncTest do
...
   alias Ecto.Changeset
 
-  describe "installation_repositories_event/1 added" do
+  @doc """
+  Some clauses defined seem difficult or impossible to reach so their tests were ommitted
+    - {:error, :validation_error_on_syncing_installation, Changeset.t()}
+    - {:error, :validation_error_on_marking_installation_processed, Changeset.t()}
+    - {:error, :unexpected_transaction_outcome, any}
+  However, if these clauses can be caused by some updates upstream we should cover them with tests
+  """
```
Comment (begedin) [inline test/lib/code_corps/github/sync/sync_test.exs]: `@doc` is used generally for public function documentation only (`def`). This should just be a simple comment:

```Elixir
# Some clauses defined seem difficult or impossible to reach so their tests were omitted
#  - {:error, :validation_error_on_syncing_installation, Changeset.t()}
#  - {:error, :validation_error_on_marking_installation_processed, Changeset.t()}
#  - {:error, :unexpected_transaction_outcome, any}
# However, if these clauses can be caused by some updates upstream we should cover them with tests
```

Also, "omitted" is spelled with one `m`

**Review by begedin — CHANGES_REQUESTED** [review]
Just a small change here and good to go 👍

File: test/lib/code_corps/github/sync/sync_test.exs (hunk 1 above)
Comment (zacck-zz) [reply test/lib/code_corps/github/sync/sync_test.exs]: fixed this!

**Review by begedin — APPROVED** [review]

**Comment by zacck-zz** [comment]
🤘

**Comment by begedin** [comment]
Merged 👍 

Thank you for your time and work @zacck

**Pull request merged**